### ISSUE TITLE
Kedro Viz Lite User Warnings

### ISSUE THREAD

**Issue by ravi-kumar-pilla** [kedro-org/kedro-viz#2058]
## Description

This ticket is to address next steps in https://github.com/kedro-org/kedro-viz/pull/1966

1. `FE notice with a warning of missing imports` suggestion from TD on Lightweight Kedro Viz implementation
2. `CLI showing a warning message with missing imports` (already [implemented](https://github.com/kedro-org/kedro-viz/pull/1966#issuecomment-2307429604) but needs confirmation from design on the wording. @stephkaiser )
3. `MemoryDataset to custom ImportErrorDataset in-case of missing dataset dependencies in the metadata panel`. Since the focus is mostly on flowchart, I am not sure if we should create a custom dataset or leave this as MemoryDataset. (need to discuss this during backlog grooming)

## Context

We mocked missing dependencies in Kedro project when users run `kedro viz --lite`. This ticket aims at letting the users aware of the mocked dependencies both in UI and CLI

## Possible Implementation

1. In CLI, have a logger warning message to inform users of the missing dependencies. Example:

```
[08/23/24 16:33:59] WARNING  Kedro-Viz has mocked the following             data_loader.py:173
                             dependencies for lite-mode.                                      
                             ['sklearn.base', 'matplotlib.pyplot',                            
                             'matplotlib', 'seaborn', 'PIL',                                  
                             'sklearn.model_selection', 'sklearn',                            
                             'sklearn.metrics']                                               
                             In order to get a complete experience of Viz,                    
                             please install the missing Kedro project                         
                             dependencies 
```

2. In UI, have a hidden banner or something similar to [this](https://docs.servicenow.com/bundle/utah-platform-user-interface/page/administer/navigation-and-ui/concept/next-experience-banner-announcements.html) website and Kedro theme

3. Leave the MemoryDataset as-is or implement a custom dataset and discuss on what all functionalities to include in the dataset (mostly leaving the bare-minimum) to show up in metadata panel

## Checklist

- [x] Include labels so that we can categorise your feature request

**Comment by ravi-kumar-pilla** [comment]
Hi @rashidakanchwala and @stephkaiser ,

Based on our discussion last week about the CLI message for missing dependencies in `--lite` mode -

Current message displays as -
```bash
[08/23/24 16:33:59] WARNING  Kedro-Viz has mocked the following             data_loader.py:173
                             dependencies for lite-mode.                                      
                             ['sklearn.base', 'matplotlib.pyplot',                            
                             'matplotlib', 'seaborn', 'PIL',                                  
                             'sklearn.model_selection', 'sklearn',                            
                             'sklearn.metrics']                                               
                             In order to get a complete experience of Viz,                    
                             please install the missing Kedro project                         
                             dependencies 
```

Should the updated message be -

```bash
[08/23/24 16:33:59] WARNING  In order to get a complete experience of Kedro-Viz,             data_loader.py:173             
                             please install the missing Kedro project dependencies below:                             
                             ['sklearn.base', 'matplotlib.pyplot',                            
                             'matplotlib', 'seaborn', 'PIL',                                  
                             'sklearn.model_selection', 'sklearn',                            
                             'sklearn.metrics']                  
```

Thank you

**Comment by stephkaiser** [comment]
Designs for the warning banners can be found here - https://www.figma.com/design/3kSpvIO1veLKfy9qHxuXwF/Kedro-WIP?node-id=2550-103078

Let me know what you think about the information provided. The 'Learn more' button can link to documentation with more information about missing dependencies.

**Comment by rashidakanchwala** [comment]
The design looks great to me. thanks @stephkaiser
